build_config/lib/src/parse.dart

```diff
@@ -322,3 +315,33 @@ List<String> _readListOfStringsOrThrow(
   }
   return new List<String>.from(value as List);
 }
+
+InputSet _readInputSetOrThrow(Map<String, dynamic> options, String option,
+    {InputSet defaultValue, bool allowNull: false}) {
+  final value = options[option];
+  if (value == null && defaultValue != null) return defaultValue;
+  if (value == null && allowNull) return null;
+
+  if (value is List && value.every((v) => v is String)) {
+    final include = new List<String>.from(value);
+    return new InputSet(include: include);
+  }
+
+  if (value is Map) {
+    final invalidOptions = value.keys.toList()
+      ..removeWhere((k) => k == 'include' || k == 'exclude');
+    if (invalidOptions.isNotEmpty) {
+      throw new ArgumentError('Got invalid options `$invalidOptions` '
+          'for $option. Only "include", and "exclude" are allowed');
+    }
+    final include = value.containsKey('include')
```

Comment:
You could simplify these a bit:

```dart
final include = (value['include'] as List<String>)?.toList()
```

Reply:
The suggested `as` won't work in Dart 2 semantics so I went with the future-proof approach.